cluster/src/lib.rs

```diff
@@ -44,6 +47,46 @@ pub type ClusterRef = Arc<dyn Cluster + Send + Sync>;
 
 pub type TableManipulatorRef = Arc<dyn TableManipulator + Send + Sync>;
 
+pub type ClusterTopologyRef = Arc<ClusterTopology>;
+
+pub type TableLocs = HashMap<String, ShardView>;
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(default)]
+pub struct SchemaConfig {
+    pub auto_create_tables: bool,
+    pub default_engine_type: String,
+    pub default_timestamp_column_name: String,
+}
+
+impl Default for SchemaConfig {
+    fn default() -> Self {
+        Self {
+            auto_create_tables: false,
+            default_engine_type: ANALYTIC_ENGINE_TYPE.to_string(),
+            default_timestamp_column_name: TIMESTAMP_COLUMN.to_string(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Node {
+    pub addr: String,
```

Comment:
Would `SocketAddr` be better?

Reply:
`SocketAddr` introduces a limit that the `addr` must be an IP, but in some cases the `addr` may be a domain name.

Reply:
got it